### ISSUE TITLE
Confused about these 3 usages

### ISSUE THREAD

**Issue by najafmurtaza** [UKPLab/sentence-transformers#359]
Hi @nreimers,

what is the difference between [Sentence Embeddings with Transformers](https://github.com/UKPLab/sentence-transformers/blob/master/docs/usage/computing_sentence_embeddings.md#sentence-embeddings-with-transformers), [Loading custom BERT models](https://github.com/UKPLab/sentence-transformers/#loading-custom-bert-models) and [Models](https://github.com/UKPLab/sentence-transformers/#models).

From my understanding, 
[Loading custom BERT models](https://github.com/UKPLab/sentence-transformers/#loading-custom-bert-models) is for models which are fine-tuned/trained using Sentence-Transformer lib.
[Sentence Embeddings with Transformers](https://github.com/UKPLab/sentence-transformers/blob/master/docs/usage/computing_sentence_embeddings.md#sentence-embeddings-with-transformers) is to directly use Hugging Face models instead of using sentence_transformer.
[Models](https://github.com/UKPLab/sentence-transformers/#models) is low level of [Loading custom BERT models](https://github.com/UKPLab/sentence-transformers/#loading-custom-bert-models)

But if we can use [Sentence Embeddings with Transformers](https://github.com/UKPLab/sentence-transformers/blob/master/docs/usage/computing_sentence_embeddings.md#sentence-embeddings-with-transformers) then whats the usage/advantage of other two?

**Comment by nreimers** [comment]
Hi @najafmurtaza 

sentence-transformers offers the following main features:
- It has different building blocks to create models that are able to produce sentence embeddings. The different building blocks are described here: https://github.com/UKPLab/sentence-transformers/#models   . The most simple network architecture is a BERT (or similar) network followed by a mean pooling. But this library also allows to add e.g. dense layers on top of it to change the embedding dimension.
- Further it contains various classes that help to fine-tune the constructed models so that they produce suitable sentence embeddings
- Finally, it implements convenience methods to efficiently compute sentence embeddings or to compare sentence embeddings with each other.



When you use a basic SentenceTransformer Model (like BERT + pooling), you are able to use it directly without sentence-transformers as described here: https://github.com/UKPLab/sentence-transformers/blob/master/docs/usage/computing_sentence_embeddings.md#sentence-embeddings-with-transformers

It is just another option.

What you loose is the possibility to easily tune your models and different helping code to make the computation of embeddings more efficient. Further, that code snippet only allows to run basic models consisting of  Transformer-Model + Pooling. If you have layers on top of it, like a dense layer to project your embeddings e.g. from 768 to only 256 dimensions, it becomes much harder to run it with transformers only package.

Best
Nils Reimers

**Comment by najafmurtaza** [comment]
What I am thinking to achieve is to use Hugging Face AlexNet/T5 model but with sentence-transformer. 

model = SetenceTransformer('T5/AlexNet')
embds = model.encode(sentences)

Is that possible?

**Comment by nreimers** [comment]
In that case you must create your model from scratch:
https://github.com/UKPLab/sentence-transformers/blob/master/docs/training/overview.md#creating-networks-from-scratch

You need a Transformer layer and a pooling layer.

Then it should work.

**Comment by najafmurtaza** [comment]
In conclusion, In order to use sentence transformer lib. either use Pre-trained models available in repo or train/finetune a new one. Right?

**Comment by nreimers** [comment]
Right. BERT and other transformer models give out-of-the-box sadly no good text representations that could be compared with cosine similarity. Hence, fine-tuning is required.
